omc3/scripts/kmod_summary.py

```diff
@@ -0,0 +1,221 @@
+"""
+Generate summary tables from K-modulation results.
+
+This function collects and summarizes K-modulation results from multiple measurement directories.
+For each beam, it generates:
+1. A DataFrame containing all imported K-modulation measurements.
+2. A text file with optinally the averaged results, which can be used to create a logbook entry.
+
+**Arguments:**
+
+*--Required--*
+
+- **meas_paths** *(Sequence[Path | str])*:
+
+    Directories of K-modulation results to import.
+    These need to be the paths to the root-folders containing B1 and B2 sub-dirs.
+
+*--Optional--*
+
+- **averaged_meas** *(dict[str, dict[int, tfs.TfsDataFrame]], default=None)*:
+        Precomputed averaged K-modulation results. If provided, these are introduced in the
+        summary table for the logbook.
+
+- **output_dir** *(Path | str, default=None)*:
+        Path to the directory where to write the output files.
+        If None, no files are written.
+
+- **logbook** *(str, default=None)*:
+        Name of the logbook to which formatted summary tables are added. If None,
+        no logbook entry is created.
+
+**Returns:**
+
+Tuple containing:
+
+1. **kmod_summaries** *(dict[str, list[tfs.TfsDataFrame]])*:
+        Mapping of beam names to lists of K-modulation summary DataFrames.
+
+2. **tables_logbook** *(dict[str, list[list[str]]])*:
+        Mapping of beam names to lists of formatted tables suitable for logbooks.
+"""
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+import tfs
+
+from omc3.optics_measurements.constants import BEAM_DIR, BETASTAR, BETAWAIST, ERR, LABEL, WAIST
+from omc3.scripts.create_logbook_entry import main
```

Comment:
This should be imported under a new name, `main(...)` later in the code will not tell me anything. Especially since it's another script's main functionality.